File: 0795-k-th-symbol-in-grammar/0795-k-th-symbol-in-grammar.java
class Solution {
    public int kthGrammar(int n, int k) {
        int cur =0;
        int left =1, right = (int)Math.pow(2,n-1);
        for(int i=0;i<n;i++){
            int mid = (right+left)/2;
            if (k<= mid){
                right = mid;
            }else{
                left = mid+1;
                cur = 1-cur;
            }
        }
        return cur;
    }
}

File: 0795-k-th-symbol-in-grammar/0795-k-th-symbol-in-grammar.py
class Solution:
    def kthGrammar(self, n: int, k: int) -> int:
        cur =0
        left = 1
        right = 2**(n-1)
        for i in range(n-1):
            mid = (right+left)//2
            if k <= mid:
                right =mid
            else:
                left = mid+1
                cur = 1-cur
        return cur
        